backend/database.js

```diff
@@ -184,34 +230,52 @@ function throwIfFileOutsideDataDir (filename: string): string {
   return filepath
 }
 
+let db: any = null
+
+if (persistence === 'sqlite') {
+  const filename = './data/groupincome.db'
+  const sqlite3 = require('sqlite3')
+
+  db = new sqlite3.Database(filename, (err) => {
+    if (err) {
+      return console.error(err.message)
+    }
+    console.log('Connected to the %s SQlite database.', filename)
+  })
+  ;(async function () {
+    await new Promise((resolve, reject) => {
+      db.run('CREATE TABLE IF NOT EXISTS Strings(key TEXT UNIQUE NOT NULL, value TEXT NOT NULL)', [], (err) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
+      })
+    })
+  })()
+}
+
 if (production || process.env.GI_PERSIST) {
   // https://github.com/isaacs/node-lru-cache#usage
   const cache = new LRU({
     max: Number(process.env.GI_LRU_NUM_ITEMS) || 10000
   })
 
   sbp('sbp/selectors/overwrite', {
-    // we cannot simply map this to readFile, because 'chelonia/db/getEntry'
-    // calls this and expects a string, not a Buffer
-    // 'chelonia/db/get': sbp('sbp/selectors/fn', 'backend/db/readFile'),
-    'chelonia/db/get': async function (filename: string) {
-      const lookupValue = cache.get(filename)
+    'chelonia/db/get': async function (key: string) {
+      const lookupValue = cache.get(key)
       if (lookupValue !== undefined) {
         return lookupValue
       }
-      const bufferOrError = await sbp('backend/db/readFile', filename)
-      if (Boom.isBoom(bufferOrError)) {
-        return null
-      }
-      const value = bufferOrError.toString('utf8')
-      cache.set(filename, value)
+      const value = await sbp('backend/db/readString', key)
+      cache.set(key, value)
```

Comment:
This API must be designed to behave in the *exact same way* regardless of which backend database type is selected (including in-memory database).

Right now the behavior is inconsistent between `fs` and `sqlite`, and possibly in-memory.

If a value is not found, `fs` will return `null`, whereas `sqlite` will return `undefined`, and I think in-memory will also return `undefined`.

We need to make sure that they all return the same thing, so that the calling APIs, if their behavior depends on the return value in any way, remains consistent when the backend database is switched.

Another thing to note, is that previously, the cache would not be set when the fs backend failed to find the file. Now it is set. This is OK, but just want to make sure that that is a conscious decision, because it does represent a minor change in behavior (that I think is fine).